Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -1780,6 +1788,40 @@ def getResultSummary(self):
         return {u'step': status}
 
 
+class DisplayUnexpectedResults(buildstep.BuildStep, AddToLogMixin):
+    name = 'display-unexpected-results'
+
+    @defer.inlineCallbacks
+    def run(self):
+        result_directory = f"public_html/results/{self.getProperty('buildername')}/{self.getProperty('archive_revision')} ({self.getProperty('buildnumber')})"
+        unexpected_results_json = os.path.join(result_directory, SCAN_BUILD_OUTPUT_DIR, 'unexpected_results.json')
+        with open(unexpected_results_json) as f:
+            unexpected_results_data = json.load(f)
+        yield self.getFilesPerProject(unexpected_results_data, 'passes')
+        yield self.getFilesPerProject(unexpected_results_data, 'failures')
+        if self.getProperty('unexpected_failing_files', 0) or self.getProperty('unexpected_passing_files', 0):
+            defer.returnValue(FAILURE)
+        defer.returnValue(SUCCESS)
```

Comment:
I don't know Buildbot Python very well, but do you need a `return` statement here so that this code doesn't call `defer.returnValue(SUCCESS)` every time?

```suggestion
        if self.getProperty('unexpected_failing_files', 0) or self.getProperty('unexpected_passing_files', 0):
            defer.returnValue(FAILURE)
            return
        defer.returnValue(SUCCESS)
```